playground/infrastructure/helper.py

```diff
@@ -44,29 +57,193 @@ def find_examples(work_dir: str) -> List[Example]:
         categories:
             - category-1
             - category-2
+    If some example contain beam tag with incorrect format raise an error.
 
     Args:
         work_dir: directory where to search examples.
+        supported_categories: list of supported categories.
 
     Returns:
         List of Examples.
     """
-    # TODO [BEAM-13135] Implement
-    pass
+    failed = False
+    examples = []
+    for root, _, files in os.walk(work_dir):
+        for filename in files:
+            filepath = os.path.join(root, filename)
```

Comment:
This function goes very deep – more than 3 indentations, it makes. it hard to read. The least we can do is to extract the processing of each file into a separate function.

Reply:
Changed.